tests/periph_spi/main.c

```diff
@@ -242,163 +267,236 @@ int cmd_bench(int argc, char **argv)
         return 1;
     }
 
-    puts("### Running some benchmarks, all values in [us] ###\n");
+    puts("### Running some benchmarks, all values in [us] ###");
+    puts("### Test\t\t\t\tTransfer time\tuser time\n");
 
     /* 1 - write 1000 times 1 byte */
+    sched_start = _sched_ticks();
     start = xtimer_now_usec();
     for (int i = 0; i < BENCH_REDOS; i++) {
         in = spi_transfer_byte(spiconf.dev, spiconf.cs, false, out);
         (void)in;
     }
     stop = xtimer_now_usec();
+    sched_stop = _sched_ticks();
     printf(" 1 - write %i times %i byte:", BENCH_REDOS, 1);
-    printf("\t\t\t%i\n", (int)(stop - start));
+    printf("\t\t\t%"PRIu32"\t%"PRIu32"\n", (stop - start),
+           xtimer_diff(sched_stop, sched_start).ticks32);
```

Comment:
Shouldn't that be `xtimer_usec_from_ticks(xtimer_diff(sched_stop, sched_start));`?

Reply:
That would also explain why the numbers on ATmega are ¼ of what is expected - there xtimer is sourced from a 250kHz clock.

Reply:
If you want to add `xtimer_diff_usec()` that would be great :wink:

Reply:
I've cleaned it up a bit and added a `xtimer_usec_from_ticks` here and there.